bindep.txt

```diff
@@ -1,2 +1,3 @@
 python3-rpm [(platform:redhat platform:base-py3)]
 rpm-python [(platform:redhat platform:base-py2)]
+python38-requests [platform:centos-8 platform:rhel-8]
```

Comment:
how does this differ to `(platform:redhat platform:base-py3)`?
also, should this be python3-requests, to allow both 3.6 and 3.8, depending on the rest of the environment?

I would prefer not to dictate what python env you use in the execution container?

Reply:
I am unsure why you are using base-py2 / base-py3, must be something specific for your tooling.

For EEs, that are build on rhel-8 / centos-8 we've decided on using python38, so that is already decided. bindep in itself is opinionated sadly, so everything today downstream will only be python38

Reply:
We don't have any tooling. Nor do we produce EEs ourself.

At some point we were pinged to add system deps, and after reading documentation of `bindep`, this seemed like the most straight forward way to say:
* on el-based distributions with py2 as the default (centos7, rhel7, oel7, etc): use rpm-pyhon
* on el-based distributions with py3 as default (el8…) use python3-rpm

Reply:
So, this is still open, and I would like to get some closure here.

If for EE, you decided to go with Python 3.8, where does the user get the `rpm` module from? There is no `python38-rpm`.
So generating an EE with Py38 won't work for all modules of this collection?

For a moment I thought we should add both `python3-requests` and `python38-requests`, so it at least also works for both Pythons, but then I remembered that 3.6 is going away soon from Ansible anyways… But it's probably still the most "correct" solution, for now.

Reply:
This is specifically for downstream packaging in brew, where we prefer to use the packaged version of python38-requests over the pip version.

If you'd like to set up those value for centos, that should be okay. For rhel-8, we do not need them, just python38-requests.